src/Discord.Net.Interactions/Extensions/IDiscordInteractionExtensions.cs

```diff
@@ -44,6 +44,44 @@ public static async Task RespondWithModalAsync<T>(this IDiscordInteraction inter
             await SendModalResponseAsync(interaction, customId, modalInfo, options, modifyModal);
         }
 
+        /// <summary>
+        ///     Respond to an interaction with a <see cref="IModal"/> and fills the value fields of the modal using the property values of the provided
+        ///     instance.
+        /// </summary>
+        /// <typeparam name="T">Type of the <see cref="IModal"/> implementation.</typeparam>
+        /// <param name="interaction">The interaction to respond to.</param>
+        /// <param name="modal"><see cref="IModal"/> instance to get field values from.</param>
+        /// <param name="options">The request options for this <see langword="async"/> request.</param>
+        /// <param name="modifyModal">Delegate that can be used to modify the modal.</param>
```

Comment:
Why is `modifyModal` after `options`? Common library behavior places options last.

Reply:
factory delegate was an afterthought and was added a good amount of time after the Modals commit so i didnt want it to break existing code bases

Reply:
This should be marked to be changed for v4. I will open an issue. Next to that, lgtm